### PR TITLE
Added Tests under operator spec

### PR DESCRIPTION
**Pull request by praveen4g0**
Updated Operator Tests in spec/operator

### CODE REVIEW

**Comment by sthaha** [comment]
@nikhil-thomas

File: spec/features/operator/clusterConfig/clusterConfig_test.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+import (
+	"testing"
+
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestClusterConfigReinstall(t *testing.T) {
+	Convey("Given the Operator is installed", t, func() {
+		Convey("Delete cluster config from cluster", nil)
```
Comment (nikhil-thomas) [inline spec/features/operator/clusterConfig/clusterConfig_test.go]: before testing effects deleting config, shouldn't we test installing operator installed pipelines, triggers and cluster tasks

Comment (nikhil-thomas) [reply spec/features/operator/clusterConfig/clusterConfig_test.go]:
```
Convey("Given the Operator is installed", t, func() {
    Convey("It should install the following", func() {
        Convey("installs Pipelines 0.8", nil)
	Convey("installs Triggers 0.1", nil)
	Convey("It should install the following", func() {
        ...
        }
    Convey("Delete cluster config from cluster", nil)
    Convey("Create new config name 'Cluster' with target namespace 'openshift-pipelines'", nil)
    Convey("It should install the following", func() {
        Convey("installs Pipelines 0.8", nil)
	Convey("installs Triggers 0.1", nil)
	Convey("It should install the following", func() {
        ...
        }
```

**Comment by nikhil-thomas** [comment]
@praveen4g0 
make sure you write better PullRequest Description and commit messages

https://chris.beams.io/posts/git-commit/

File: spec/features/operator/clusterConfig/clusterConfig_test.go (hunk 1 above)
Comment (praveen4g0) [reply spec/features/operator/clusterConfig/clusterConfig_test.go]: Sure, But there is one more test which Does this https://github.com/praveen4g0/release-tests/blob/master/spec/install/olm/install_test.go#L9, I hope this part can be skipped here, it leads to duplication any how after Reinstall We need to check those

Comment (nikhil-thomas) [reply spec/features/operator/clusterConfig/clusterConfig_test.go]: i see :+1:

File: spec/features/operator/clusterConfig/clusterConfig_test.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+		Convey("Verify pipeline, Triggers, ClusterTasks has been uninstalled or not", nil)
+		Convey("Create new config name 'Cluster' with target namespace 'openshift-pipelines'", nil)
```
Comment (sthaha) [inline spec/features/operator/clusterConfig/clusterConfig_test.go]: `It ` below should move inside this block

File: spec/features/operator/rbac/rbac_triggers_test.go [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import (
+	"testing"
+
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestTriggersSA(t *testing.T) {
+	Convey("Given the Operator is installed", t, func() {
+		Convey("create a new namespace", func() {
+			Convey("Then I Create a new Triggers SA with Required RBAC", nil)
```
Comment (sthaha) [inline spec/features/operator/rbac/rbac_triggers_test.go]: Could you rephrase this? 
The test seems quite confusing

File: spec/features/operator/triggers/triggers_Install_test.go [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package triggers
+
+import (
```
Comment (sthaha) [inline spec/features/operator/triggers/triggers_Install_test.go]: could you please keep the filenames to lowercase and just call it `install_test.go` since it is already inside `triggers` directory?

File: spec/features/pipelines/pipelines_test.go [hunk 5]
```diff
@@ -7,8 +7,22 @@ import (
...
 func TestRunTutorial(t *testing.T) {
-
 	Convey("Given that Operator is installed", t, func() {
 		Convey("I should be able to run Pipelines Tutorial as a non admin", nil)
 	})
 }
+
+func TestPipelineRunForClusterTasks(t *testing.T) {
+	Convey("Given that Operator is installed", t, func() {
+		Convey("I should be able to run pipeline successfuly", func() {
```
Comment (sthaha) [inline spec/features/pipelines/pipelines_test.go]: this needs to change as well to follow ... 

Given
When
Then 

Instead of 
Given 
then
When

File: spec/features/tasks/use_tasks_test.go [hunk 6]
```diff
@@ -8,11 +8,10 @@ import (
 
 func TestUseClusterTasks(t *testing.T) {
 	Convey("Given that Operator is installed", t, func() {
-
 		Convey("When I am logged in as a non-admin user", func() {
+			Convey("I Create a new namespace", nil)
```
Comment (sthaha) [inline spec/features/tasks/use_tasks_test.go]: why is this step even needed?

Comment (sthaha) [reply spec/features/tasks/use_tasks_test.go]: Listing and creating pipelines are 2 differnet tests

File: spec/features/operator/clusterConfig/clusterConfig_test.go (hunk 2 above)
Comment (praveen4g0) [reply spec/features/operator/clusterConfig/clusterConfig_test.go]: Done

File: spec/features/operator/rbac/rbac_triggers_test.go (hunk 3 above)
Comment (praveen4g0) [reply spec/features/operator/rbac/rbac_triggers_test.go]: Makes sense Created a new Test for that

File: spec/features/operator/triggers/triggers_Install_test.go (hunk 4 above)
Comment (praveen4g0) [reply spec/features/operator/triggers/triggers_Install_test.go]: done

File: spec/features/tasks/use_tasks_test.go (hunk 6 above)
Comment (praveen4g0) [reply spec/features/tasks/use_tasks_test.go]: makes sense created 2 tests

File: spec/features/operator/clusterconfig/clusterconfig_test.go [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+import (
+	"testing"
+
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestClusterConfigReinstall(t *testing.T) {
+	Convey("Given the Operator is installed", t, func() {
+		Convey("Delete cluster config from cluster", nil)
+		Convey("Verify pipeline, Triggers, ClusterTasks has been uninstalled or not", nil)
```
Comment (vdemeester) [inline spec/features/operator/clusterconfig/clusterconfig_test.go]: we can remove `or not`, as *it should be removed*, if it's not it's a failure

File: spec/features/operator/curatedtasks/clustertask_test.go [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+import (
+	"testing"
+
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestClusterTasksValidation(t *testing.T) {
+	Convey("Given the Operator is installed", t, func() {
+		Convey("When I logged in as non-admin user", func() {
+			Convey("Then I should be able to validate ClusterTask manifest yamls ", nil)
```
Comment (vdemeester) [inline spec/features/operator/curatedtasks/clustertask_test.go]: By `validate` do we mean "view" or do we mean "create a ClusterTask1 ?

Comment (vdemeester) [reply spec/features/operator/curatedtasks/clustertask_test.go]: or create a `TaskRun` using that `ClusterTask` and having it work ?

File: spec/features/operator/pipeline/install_test.go [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+
+import (
+	"testing"
+
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestPipelinesInstall(t *testing.T) {
+	Convey("Given the Operator is installed", t, func() {
+		Convey("It should install pipelines controllers as a deployment to target namespace (openshift-pipelines)", nil)
```
Comment (vdemeester) [inline spec/features/operator/pipeline/install_test.go]: We could rephrase this into `it should have installed …` right ? as we will "only" verify that the thing are present, aka installed.

Comment (praveen4g0) [reply spec/features/operator/pipeline/install_test.go]: right it should be``` it should have installed```

File: spec/features/operator/curatedtasks/clustertask_test.go (hunk 8 above)
Comment (praveen4g0) [reply spec/features/operator/curatedtasks/clustertask_test.go]: it's lint tests actually, TaskRun, PipelineRun using ClusterTask is already defined in another test
